### ISSUE TITLE
question about training time

### ISSUE THREAD

**Issue by wmWangmiao** [tulerfeng/Video-R1#97]
Hi! 👋
Thank you for sharing this amazing project!

I’m currently trying to reproduce the training process using 8×H800 GPUs, but it’s running significantly slower than expected. I’ve followed the provided instructions and used the same training parameters as in the configuration (e.g., batch size, learning rate, etc.). However, the training speed seems much lower than what might be intended.

For reference, here are the training parameters I used:

```
cd src/r1-v

export DEBUG_MODE="true" # Enable Debug if you want to see the rollout of model during RL
export LOG_PATH="./debug_log_2b.txt"

# For resume training:  --resume_from_checkpoint Model_Path \
# Set temporal to choose between T-GRPO and GRPO, and len_control to enable or disable the length control reward.

# Qwen/Qwen2.5-VL-7B-Instruct

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 torchrun --nproc_per_node="8" \
    --nnodes="1" \
    --node_rank="0" \
    --master_addr="127.0.0.1" \
    --master_port="12365" \
    src/open_r1/grpo.py \
    --output_dir "./log/Qwen2.5-VL-7B-GRPO" \
    --model_name_or_path '/data/wm/Video-R1/src/r1-v/log/Qwen2.5-VL-7B-Video-7B-cot-sft-2' \
    --dataset_name "./Video-R1-data/Video-R1-260k.json" \
    --deepspeed local_scripts/zero3.json \
    --max_prompt_length 16384 \
    --max_completion_length 768 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --learning_rate 1e-6 \
    --lr_scheduler_type "cosine" \
    --weight_decay 0.01 \
    --bf16 True\
    --logging_steps 1 \
    --gradient_checkpointing true \
    --temporal true \
    --len_control true \
    --attn_implementation flash_attention_2 \
    --max_pixels 401408 \
    --num_train_epochs 1 \
    --run_name Video-R1-GRPO
    --save_steps 100 \
    --beta 0.04 \
    --max_grad_norm 5 \
    --save_only_model false \
    --num_generations 8  # number of outputs G in grpo, reduce it would lead to faster training and smaller memory cost but higher variance  
```

And here’s the current speed observation:

<img width="2800" height="1643" alt="Image" src="https://github.com/user-attachments/assets/83318e78-c80f-4df3-98ac-e0544f6230a4" />

I’d really appreciate it if you could let me know whether anything in my setup seems off, or if you have any suggested optimizations to improve the training speed.

**Comment by tulerfeng** [comment]
Thanks for your questions. As stated in the paper, we only conduct RL training for 1000 steps.

Therefore, you don't need to train for one epoch, just training over 1000 steps and used the saved checkpoints at 1000 steps

**Comment by wmWangmiao** [comment]
> Thanks for your questions. As stated in the paper, we only conduct RL training for 1000 steps.
> 
> Therefore, you don't need to train for one epoch, just training over 1000 steps and used the saved checkpoints at 1000 steps

Hi @tulerfeng !
Thank you so much for your clear explanation!  It’s greatly appreciated and clears up my confusion.
